src/enforcer.rs

```diff
@@ -181,8 +181,14 @@ impl Enforcer {
         let e_ast = get_or_err!(self, "e", ModelError::E, "effector");
 
         for (i, token) in r_ast.tokens.iter().enumerate() {
-            let scope_exp = format!("let {} = \"{}\";", token, rvals[i].as_ref());
-            engine.eval_with_scope::<()>(&mut scope, &scope_exp)?;
+            if token == "r_obj" && &rvals[i].as_ref()[0..=0] == "{" {
```

Comment:
I believe it's not only for `r_obj` because in the doc we can also have something like:

```rust
r.sub.Domain = r.obj.Domain
```

Reply:
I think we should also validate the input to make sure it's a `Map`

Reply:
You are right, I found the doc at https://casbin.org/docs/en/abac#how-to-use-abac

Check valid json depends on third party json library, may be a little heavy, we can let the users to make sure they are passing a valid json string. A little concern is that a json string may be conflict with normal resources, but we don't need to worry too much because in ABAC we can't use policy, would reduce conflict with other model.